### ISSUE TITLE
Checkboxes and Radios - target size is small

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly#865]
For accessibility we'd like to have a minimum size of 32px and a goal of 44px for elements that are clickable. Currently, checkboxes measure at 14x14 px and radios at 14x15 px.

**Comment by mcarrano** [comment]
@mcoker @LHinson @rachael-phillips Was this ever addressed? If not, it's something that we should get on the roadmap.

**Comment by mcoker** [comment]
@mcarrano no, we're still using the browser defaults. FWIW, I created a quick demo of the direction we might go with it here - https://codepen.io/mcoker/pen/WBoPab

**Comment by mcarrano** [comment]
Thanks for that update @mcoker . I will add this to our design backlog so we can give it some consideration. I know that an intermediate state checkbox is also something that has been asked for so it would be good to address all of these at the same time.

Regarding the Codepen demo, these elements seem exceedingly large. @jgiardino would you really expect something this large on a form, for example? I know that 44px is what Apple recommends for touch targets on iPhone apps, but not sure this is appropriate for typical use cases that we have. Is the a11y issue about visibility, touch target or a little bit of both?

**Comment by bkjeffer** [comment]
I've done a bit of research to see how other design systems handle checkboxes and radio buttons. From what I gathered, more systems were using sizes 16x16 and below, though there were a few notable exceptions. I've created a document (based on the design systems I viewed) that list the sizes of the checkboxes and radio buttons in descending order, along with links to view the larger implementations. https://docs.google.com/document/d/1M-mwQsvUbCXOVnlD9U3yqdKhGo5VXDALoHWmCIR_20I/edit?usp=sharing

A few things to consider: when we say we want to update "for accessibility", are we primarily focusing on how things appear on smaller screens (as opposed to considering users who may be impaired in some way)? And if the appearance on smaller screens is our focus, have we considered responsive design elements, so that the clickable elements would (only) show bigger when on smaller screens? Are we sure that using larger checkboxes and radio buttons would be more useful than implementing different components (e.g. switches and tabs) that often are used for smaller screens?

**Comment by jgiardino** [comment]
Thanks for pulling all of this information together, @bkjeffer! It's great to see what other design systems have.  

> A few things to consider: when we say we want to update "for accessibility", are we primarily focusing on how things appear on smaller screens (as opposed to considering users who may be impaired in some way)?

These are some of the users that benefit from increased target size:
_(By "target size" I mean the area on the screen that the user can click to toggle these inputs. But this does not have to be the visual size.)_

- Users with impaired mobility using a mouse - For these users, the smaller the target size is, the more difficult it is to hit that target.
- Users using touch - Similar issue as above, but these users might also benefit from increased spacing between adjacent clickable elements. For these users smaller target sizes coupled with spacing that's too small makes it more difficult to target the desired input. This user is not necessarily synonymous with the mobile user, given the number of larger screen sizes that support touch.



> Regarding the Codepen demo, these elements seem exceedingly large. @jgiardino would you really expect something this large on a form, for example?

I expect something a little closer to what [Material UI](https://material-components.github.io/material-components-web-catalog/#/component/checkbox) does. It's really hard to see in the example below, but when you hover over the checkbox, the circle that displays on hover is 40x40px, and the user can click anywhere in that 40x40px area, but the checkbox itself is more of a standard size visually. 

![image](https://user-images.githubusercontent.com/21063328/59284366-23e96180-8c3a-11e9-8b29-fcca57dfc368.png)

Something else I like about their solution is that it doesn't require a `<label>` to increase the target size, since we have a lot of patterns (like datalist and tables) that include checkboxes without `<label>` elements. 

@mcoker What are your thoughts on the Material UI solution for checkboxes? Is something like this doable in all of our patterns that use checkboxes/radio buttons?

**Comment by mcarrano** [comment]
Going to kick this back to the design backlog for visual design attention as next step.

**Comment by mcarrano** [comment]
@jessiehuff can you take another look at this one. Is this something you think we still need to address for meeting accessibility goals?

**Comment by jgiardino** [comment]
> Is this something you think we still need to address for meeting accessibility goals?

I would say no.

There aren't any requirements in WCAG at this time related to target size of checkboxes. The only item is 2.5.5: Target Size, which is AAA (whereas our goal is AA). In their [explanation of this item](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) includes the following section: 
> User Agent Control: If the size of the target is not modified by the author through CSS or other size properties, then the target does not need to meet the target size of 44 by 44 CSS pixels. 

I would argue that native html checkboxes qualify as a "target is not modified by the author through CSS or other size properties". So even in the AAA criteria related to target size, this isn't something we would be expected to address.
